gateway/ln-gateway/src/lib.rs

```diff
@@ -1240,8 +1240,8 @@ impl Gateway {
                 warn!("Failed to load client for federation: {federation_id}");
             }
 
-            if config.mint_channel_id > next_channel_id {
-                next_channel_id = config.mint_channel_id + 1;
+            if scid > next_channel_id {
+                next_channel_id = scid + 1;
```

Comment:
Doing the "max used channel id" calculation in this loop isn't very readable. I'd much prefer iterating over all configs a second time to determine the next channel id to be given out:

```rust
let next_channel_id = configs.iter()
    .map(|cfg| cfg.mint_channel_id)
    .max() + 1;
```

After writing this I also think the current code is subtly broken: if `scid == next_channel_id` then `next_channel_id` won't be incremented even though the SCID is already in use.

Reply:
ah yes. Fixed this by track just the max used scid. we only need increment the value when connecting new federations